### PR TITLE
Playlist file handling

### PR DESCRIPTION
**Pull request by wolkenschieber**
Hi,

I saw you wrote a spotify ripper using the pyspotify 2 library. I decided to port my playlist handling feature from my old ripper to your new tool. The songlibrary scans the download dir for mp3 files and matches them with the newly ripped ones. Further in playlist mode, the idx/targetfile name is updated and missing entries are moved into the attic.

I used Python3 for testing so there you be one or two places which might nor work with Python 2.

Sorry, I'm not very good with Python.

Best regrads
wolkenschieber

### CODE REVIEW

**Comment by wolkenschieber** [comment]
Obviously I'm also not very good with git :-/

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+
+### Python template
+# Byte-compiled / optimized / DLL files
+__pycache__/
 *.py[cod]
 
 # C extensions
 *.so
 
+<<<<<<< HEAD
```
Comment (OmgImAlexis) [inline .gitignore]: This is from the merge, you should go through the files and have a look for anything with ``<<<<<<< HEAD`` and fix them up.

Comment (wolkenschieber) [inline .gitignore]: This only affects the .gitignore and the file isn't critical.

However my fork and this project diverged, so this pull request would require some refactoring.

**Comment by jrnewell** [comment]
I sort of met you halfway with this PR and implemented the ``--playlist-sync`` option in version ``2.4.0``.  It will sync your local ripped files with a playlist (i.e rename, remove or add).  That way you can reorder your playlist or whatever and the filename index will reflect that.

**Pull request closed without merging**